test/base_test_fixture.h

```diff
@@ -871,30 +871,44 @@ struct base_test_fixture
 
         check_rows_equal(execute(connection, query), 10);
 
+        REQUIRE(connection.transactions() == 0);
         {
             nanodbc::transaction transaction(connection);
+            REQUIRE(connection.transactions() == 1);
             execute(connection, NANODBC_TEXT("delete from transaction_test;"));
             check_rows_equal(execute(connection, query), 0);
+            REQUIRE(connection.transactions() == 1);
             // ~transaction() calls rollback()
         }
+        REQUIRE(connection.transactions() == 0);
 
         check_rows_equal(execute(connection, query), 10);
 
+        REQUIRE(connection.transactions() == 0);
         {
             nanodbc::transaction transaction(connection);
+            REQUIRE(connection.transactions() == 1);
             execute(connection, NANODBC_TEXT("delete from transaction_test;"));
             check_rows_equal(execute(connection, query), 0);
-            transaction.rollback();
+            REQUIRE(connection.transactions() == 1);
+            transaction.rollback(); // only requests rollback performed in ~transaction()
+            REQUIRE(connection.transactions() == 1); // transaction not released yet
```

Comment:
> A transaction that is destroyed without first having commit() called on it will default to
> rolling back, so there's no need to manually call rollback()
> that I can think of off the top of my head.

@lexicalunit I totally agree the `rollback` is redundant. Since it might be confusing (my notes on symmetry above), it could be removed or marked as deprecated. Shall we have a separate PR?